tests/functional/Glpi/Inventory/Inventory.php

```diff
@@ -7851,4 +7852,153 @@ public function testStatusIfInventoryOnAddUpdate()
+                'value' => '#0l1234567890',
+            ],
+            "xml_fields" => [
+                'name' => 'lognorma'
+            ],
+            'result' => [
+                'users_id'    => $normal_id,
+            ]
+        ];
+        yield [
+            'rules_fields' => [
+                'name'    => 'Assign user by email',
+                'action' => '_affect_user_by_email_and_regex',
+                'value' => '#0i@teclib.com',
+            ],
+            "xml_fields" => [
+                'name' => 'logglp'
+            ],
+            'result' => [
+                'users_id'    => $glpi_id,
+            ]
+        ];
+        yield [
+            'rules_fields' => [
+                'name'    => 'Assign user by sync field',
+                'action' => '_affect_user_by_sync_field_and_regex',
+                'value' => '#0r@toto',
+            ],
+            "xml_fields" => [
+                'name' => 'log_test_use'
+            ],
+            'result' => [
+                'users_id'    => $test_id,
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider getAssignUserByFieldAndRegexRules
+     */
+    public function testAssignUserByFieldAndRegex($rules_fields, $xml_fields, $result)
+    {
+        global $DB;
+
+        //create rule
+        $input_rule = [
+            'is_active' => 1,
+            'name'      => $rules_fields['name'],
+            'match'     => 'AND',
+            'sub_type'  => 'RuleAsset',
+            'condition' => \RuleAsset::ONADD + \RuleAsset::ONUPDATE
+        ];
+
+        $rule = new \Rule();
+        $rules_id = $rule->add($input_rule);
+        $this->integer($rules_id)->isGreaterThan(0);
+
+        //create criteria
+        $input_criteria = [
+            'rules_id'  => $rules_id,
+            'criteria'      => 'contact',
+            'condition' => \Rule::REGEX_MATCH,
+            'pattern' => '/^log([^@]+)/'
+        ];
+        $rule_criteria = new \RuleCriteria();
+        $rule_criteria_id = $rule_criteria->add($input_criteria);
+        $this->integer($rule_criteria_id)->isGreaterThan(0);
+
+        //create action
+        $input_action = [
+            'rules_id'  => $rules_id,
+            'action_type' => 'regex_result',
+            'field' => $rules_fields['action'],
+            'value' => $rules_fields['value']
+        ];
+        $rule_action = new \RuleAction();
+        $rule_action_id = $rule_action->add($input_action);
+        $this->integer($rule_action_id)->isGreaterThan(0);
+
+        $xml_source = "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>
+        <REQUEST>
+        <CONTENT>
+          <HARDWARE>
+            <NAME>glpixps</NAME>
+            <UUID>25C1BB60-5BCB-11D9-B18F-5404A6A534C4</UUID>
+          </HARDWARE>
+          <BIOS>
+            <MSN>640HP72</MSN>
+          </BIOS>
+          <USERS>
+            <DOMAIN>TECLIB</DOMAIN>
+            <LOGIN>" . $xml_fields['name'] . "</LOGIN>
+          </USERS>
```

Comment:
Can you add another couple of data (```<USER>``` / ```<DOMAIN>```) to test with another criteria like ```Rule::PATTERN_NOT_CONTAIN``` 

```xml
          <USERS>
            <DOMAIN>TECLIB</DOMAIN>
            <LOGIN>" . $xml_fields['name'] . "</LOGIN>
          </USERS>
          <USERS>
            <DOMAIN>TECLIB</DOMAIN>
            <LOGIN>Admin</LOGIN>
          </USERS>
```
Criteria => ```Rule::PATTERN_NOT_CONTAIN``` ```Admin```